### PR TITLE
feat(core): fix edge cases for f64 to i64 conversion

### PR DESCRIPTION
**Pull request by sarah-ek**
### Description
fixes the edge case in the vectorized f64 to i64 conversion

### Checklist 

(Use '[x]' to check the checkboxes, or submit the PR and then click the checkboxes)

* [x] Tests for the changes have been added (for bug fixes / features)
* [x] Docs have been added / updated (for bug fixes / features)
* [ ] ~~The PR description links to the related issue (to link an issue, use '#XXX'.)~~
* [x] The tests on AWS have been launched and are successful (comment with @slab-ci cpu_test and/or @slab-ci gpu_test to trigger the tests)
* [ ] ~~The draft release description has been updated~~
* [x] Check for breaking changes (including serialization changes) and add them to commit message following the conventional commit [specification][conventional-breaking]

[conventional-breaking]: https://www.conventionalcommits.org/en/v1.0.0/#commit-message-with-description-and-breaking-change-footer

### CODE REVIEW

**Comment by sarah-ek** [comment]
@slab-ci cpu_test

**Comment by sarah-ek** [comment]
@slab-ci cpu_test

File: concrete-core/src/backends/fft/private/math/fft/x86.rs [hunk 1]
```diff
@@ -30,7 +30,7 @@ pub unsafe fn mm256_cvtpd_epi64(x: __m256d) -> __m256i {
     let bits = _mm256_castpd_si256(x);
     // mask that covers the first 52 bits
     let mantissa_mask = _mm256_set1_epi64x(0xFFFFFFFFFFFFF_u64 as i64);
-    // mask that covers the 53rd bit
+    // mask that covers the 52nd bit
     let explicit_mantissa_bit = _mm256_set1_epi64x(0x10000000000000_u64 as i64);
```
Comment (IceTDrinker) [inline concrete-core/src/backends/fft/private/math/fft/x86.rs:34]: this is still 53rd, unless I'm mistaken

Comment (sarah-ek) [reply concrete-core/src/backends/fft/private/math/fft/x86.rs:34]: no,  the old comment was wrong. it's equal to 1 << 52

Comment (IceTDrinker) [reply concrete-core/src/backends/fft/private/math/fft/x86.rs:34]: ah nevermind depends on whether you count from 0 or 1

**Review by IceTDrinker — APPROVED** [review]
Thanks a lot for this Sarah 🙂
